### PR TITLE
GEN-1673 - refact(memberReviews): unify api

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

Now that _trustpilotData_ and _productReviewsData_ belong to the same feature - `memberReviews` - I've created a single API for `memberReviews` that describes what kind of data should be expected while working with reviews. `trustpilot` and `productReviews` data sources then are now compliant with that API which makes the UI simpler as no parsing is need on the client.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3815?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3809** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3809?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3810** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3810?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3813** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3813?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3814** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3814?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #3815** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3815?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3815?utm_source=stack-comment).

File: apps/store/src/components/ProductReviews/ProductReviews.tsx [hunk 1]
```diff
@@ -104,26 +102,6 @@ const useGetReviewsData = () => {
   return getReviewsData
 }
 
-const parseProductReviews = (productReviews: Array<ProductReview>): Array<Review> => {
```
Comment (robinandeer) [inline apps/store/src/components/ProductReviews/ProductReviews.tsx:107]: very nice to skip this!

**Review by robinandeer — APPROVED** [review]
looks a lot cleaner now so i like it!

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Jan 5, 10:58 AM**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3815).

**Pull request merged**